dataland-frontend/src/components/forms/CreateSfdrDataset.vue

```diff
@@ -0,0 +1,308 @@
+    AddressFormField,
+    RadioButtonsFormField,
+    YesNoNaFormField,
+    PercentageFormField,
+    ProductionSitesFormField,
+    MostImportantProductsFormField,
+    ProcurementCategoriesFormField,
+    UploadReports,
+  },
+  directives: {
+    tooltip: Tooltip,
+  },
+  emits: ["datasetCreated"],
+  data() {
+    return {
+      formId: "createSFDRForm",
+      waitingForData: true,
+      dataDate: undefined as Date | undefined,
+      companyAssociatedSfdrData: {} as CompanyAssociatedDataSfdrData,
+      sfdrDataModel: sfdrDataModel,
+      route: useRoute(),
+      message: "",
+      smoothScroll: smoothScroll,
+      uploadSucceded: false,
+      postSfdrDataProcessed: false,
+      messageCounter: 0,
+      checkCustomInputs,
+      documents: new Map() as Map<string, DocumentToUpload>,
+      referencedReportsForPrefill: {} as { [key: string]: CompanyReport } | undefined,
+      namesOfAllCompanyReportsForTheDataset: [] as string[],
+    };
+  },
+  computed: {
+    yearOfDataDate: {
+      get(): string {
+        const currentDate = this.companyAssociatedSfdrData.data?.general?.general?.fiscalYearEnd;
+        if (currentDate === undefined) {
+          return "";
+        } else {
+          const currentDateSegments = currentDate.split("-");
+          return currentDateSegments[0] ?? new Date().getFullYear();
+        }
+      },
+      set() {
+        // IGNORED
+      },
+    },
+    subcategoryVisibility(): Map<Subcategory, boolean> {
+      const map = new Map<Subcategory, boolean>();
+      for (const category of this.sfdrDataModel) {
+        for (const subcategory of category.subcategories) {
+          map.set(
+            subcategory,
+            subcategory.fields.some((field) => field.showIf(this.companyAssociatedSfdrData.data)),
+          );
+        }
+      }
+      return map;
+    },
+  },
+  props: {
+    companyID: {
+      type: String,
+      required: true,
+    },
+  },
+  created() {
+    const dataId = this.route.query.templateDataId;
+    if (dataId && typeof dataId === "string") {
+      void this.loadSfdrData(dataId);
+    } else {
+      this.waitingForData = false;
+    }
+  },
+  methods: {
+    /**
+     * Loads the SFDR-Dataset identified by the provided dataId and pre-configures the form to contain the data
+     * from the dataset
+     * @param dataId the id of the dataset to load
+     */
+    async loadSfdrData(dataId: string): Promise<void> {
+      this.waitingForData = true;
+      const sfdrDataControllerApi = await new ApiClientProvider(
+        assertDefined(this.getKeycloakPromise)(),
+      ).getSfdrDataControllerApi();
+
+      const dataResponse = await sfdrDataControllerApi.getCompanyAssociatedSfdrData(dataId);
+      const sfdrResponseData = dataResponse.data;
+      this.referencedReportsForPrefill = sfdrResponseData.data.general.general.referencedReports;
+      this.companyAssociatedSfdrData = objectDropNull(sfdrResponseData) as CompanyAssociatedDataSfdrData;
```

Comment:
IntelliJ shows me a type error here.

Compare this to the same code in "CreateLksgDataset.vue". There we have fixed it.   Use the same approach here.

Reply:
Changed